gradio/components.py

```diff
@@ -710,14 +733,21 @@ def set_interpret_parameters(
         self.interpretation_delta_type = delta_type
         return self
 
-    def get_interpretation_neighbors(self, x: float) -> Tuple[List[float], Dict]:
-        x = float(x)
+    def get_interpretation_neighbors(self, x: float | int) -> Tuple[List[float], Dict]:
+        x = self.round_to_precision(x, self.precision)
         if self.interpretation_delta_type == "percent":
             delta = 1.0 * self.interpretation_delta * x / 100
         elif self.interpretation_delta_type == "absolute":
             delta = self.interpretation_delta
         else:
             delta = self.interpretation_delta
+        if self.precision == 0 and math.floor(delta) != delta:
+            raise ValueError(
+                f"Delta value {delta} is not an . Cannot generate valid set of neighbors. "
```

Comment:
typo probably:
`
f"Delta value {delta} is not an .`

Reply:
Thank you for pointing this out!